### PR TITLE
False positive FutureReturnValueIgnored in java Standard Library 

### LINKED ISSUE #883: False positive FutureReturnValueIgnored in java Standard Library
From the javadocs of [ScheduledExecutorService#scheduleAtFixedRate](https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleAtFixedRate-java.lang.Runnable-long-long-java.util.concurrent.TimeUnit-):
> ##### Returns:
>a ScheduledFuture representing pending completion of the series of repeated tasks. The future's get() method will never return normally, and will throw an exception upon task cancellation or abnormal termination of a task execution.

So perhaps an exception for `ScheduledFuture` Futures would be nice?

### PR DESCRIPTION
**Pull request by madrob**
fixes #883

### CODE REVIEW

File: core/src/main/java/com/google/errorprone/bugpatterns/FutureReturnValueIgnored.java [hunk 1]
```diff
@@ -110,6 +111,11 @@
           instanceMethod()
               .onDescendantOf("io.netty.channel.ChannelFuture")
               .withNameMatching(Pattern.compile("addListeners?")),
+          // ScheduledExecutorService.scheduleAtFixedRate represents a series of tasks where it
+          // doesn't make sense to check for the return value
+          instanceMethod()
+              .onDescendantOf(ScheduledExecutorService.class.getName())
+              .withNameMatching(Pattern.compile("schedule(At|With)Fixed(Rate|Delay)")),
```
Comment (Stephan202) [inline core/src/main/java/com/google/errorprone/bugpatterns/FutureReturnValueIgnored.java]: :thinking: but the `Future` can be used to cancel the scheduling. (Though most users perhaps shut down the whole service instead. I wonder which of those approaches prevails within the Google codebase.)

Comment (madrob) [reply core/src/main/java/com/google/errorprone/bugpatterns/FutureReturnValueIgnored.java]: The future can be kept and used for things, of course, but in the specific case of `SES.scheduleAtFixedRate` it's arguably not a mistake to discard the return value. Quoting from the original issue filed:

> The future's get() method will never return normally

So unless you explicitly plan to cancel the future at some point, there's no reason to keep it around. You may want to keep it around to monitor the task, I guess, and restart it if it fails?

Comment (cushon) [reply core/src/main/java/com/google/errorprone/bugpatterns/FutureReturnValueIgnored.java:118]: I left a note in https://github.com/google/error-prone/issues/883#issuecomment-1724432418

**Comment by cushon** [comment]
Closing based on https://github.com/google/error-prone/issues/883#issuecomment-1724432418

**Pull request closed without merging**